### PR TITLE
Added out_sharding to jax.random.maxwell sampler

### PR DESCRIPTION
**Pull request by vfdev-5**
Description:
- Added out_sharding to jax.random.maxwell sampler
- Added a test
- Updated docstrings for shape and dtype

### CODE REVIEW

File: jax/_src/random/core.py [hunk 1]
```diff
@@ -2517,13 +2528,14 @@ def maxwell(key: ArrayLike,
     raise ValueError(f"dtype argument to `maxwell` must be a float "
                      f"dtype, got {dtype}")
   shape = core.canonicalize_shape(shape)
-  return _maxwell(key, shape, dtype)
+  out_sharding = canonicalize_sharding_for_samplers(out_sharding, "maxwell", shape)
```
Comment (jakevdp) [inline jax/_src/random/core.py:2531]: Since `_maxwell` calls `normal` with `(*shape, 3)`, I think this `out_sharding` may be incorrect

**Review by jakevdp — CHANGES_REQUESTED** [review]

File: jax/_src/random/core.py (hunk 1 above)
Comment (vfdev-5) [reply jax/_src/random/core.py:2531]: @jakevdp I may misunderstand your point, but the axis for 3 is the last one and it will be replicated and finally we compute the norm on this axis, so the output will have the shape without 3 and have `out_sharding`.
For example:
```python
import jax
import jax.numpy as jnp
import jax.random as jr

jax.config.update("jax_platforms", "cpu")
jax.config.update("jax_num_cpu_devices", 8)

key = jr.key(1)
mesh = jax.make_mesh((4, 2), ("a", "b"))
jax.set_mesh(mesh)

x = jr.maxwell(
    key,
    shape=(8, 4),
    out_sharding=jax.P("a", "b"),
)

print(jax.typeof(x), x.sharding)
# float32[8@a,4@b] NamedSharding(mesh=Mesh('a': 4, 'b': 2, axis_types=(Explicit, Explicit)), spec=P('a', 'b'), memory_kind=device)
```
What do you think?

Comment (jakevdp) [reply jax/_src/random/core.py:2531]: Thanks, that seems OK.

**Review by jakevdp — APPROVED** [review]
Thanks!

File: jax/_src/random/core.py [hunk 2]
```diff
@@ -2517,13 +2528,14 @@ def maxwell(key: ArrayLike,
...
+  return _maxwell(key, shape, dtype, out_sharding)
 
 
-@jit(static_argnums=(1, 2))
-def _maxwell(key, shape, dtype) -> Array:
+@jit(static_argnums=(1, 2, 3))
+def _maxwell(key, shape, dtype, out_sharding) -> Array:
   shape = shape + (3,)
-  norm_rvs = normal(key=key, shape=shape, dtype=dtype)
+  norm_rvs = normal(key=key, shape=shape, dtype=dtype, out_sharding=out_sharding)
```
Comment (yashk2810) [inline jax/_src/random/core.py:2538]: It would be nice if we extended the spec here to do the following:

```
new_partitions = (*out_sharding.spec, None)
out_sharding = out_sharding.update(spec=out_sharding.spec.update(partitions=new_partitions))
```

Comment (yashk2810) [reply jax/_src/random/core.py:2538]: This means we don't depend on implicit `None`'s being added which is an implementation detail.

The code here should handle that instead of depending on something downstream.

Comment (yashk2810) [reply jax/_src/random/core.py:2538]: https://github.com/jax-ml/jax/pull/37832 should do that
